src/apps/relay/dbdrivers/dbd_mysql.c

```diff
@@ -182,9 +182,9 @@ static Myconninfo *MyconninfoParse(char *userdb, char **errmsg) {
       } else if (!strcmp(s, "secret")) {
         co->password = strdup(seq + 1);
       } else if (!strcmp(s, "port")) {
-        co->port = (unsigned int)atoi(seq + 1);
+        co->port = (uint16_t)atoi(seq + 1);
```

Comment:
How does string parsing manifest in failure?

Make sure that an invalid entry aborts the operation as soon as possible. Or at least properly communicates to the caller.